Use githubs gitignore files.

Pfeil opened the pull request:
I'd prefer to use this kind of gitignore file. It uses some of the well-maintained lists from https://github.com/github/gitignore. This makes basically two changes:

- Include a lot of files we have never seen and usually will not see.
- Exclude all folders named "build" by default, with the exceptions of such folders located in `/src`. If a legit exception has to be added, it can be done like this: `!/setup/build/env.sh` but to be honest I do not expect this to happen.

I left in some things I was not able to tell where it is coming from, namely `/.settings/` and `/config/`, so no lines or files were left out in the end.

VolkerHartmann commented on .gitignore:
```diff
@@ -1,9 +1,60 @@
+# Ignore other meta folders (e.g. by netbeans)
 /.settings/
 /config/
-**/application.properties
```
I miss the following line in the new file. 
This should be also part of .gitignore as application.properties may contain private information.
(e.g. password for database, ...)
###  SPRING BOOT ###
# Ignore local settings for spring boot
# If you want to provide default settings use 'application.properties.default' and rename it while building service.
**/application.properties
##############

Pfeil commented:
Sorry for the force push, I thought I did not have a pull request already. But well.

- modified the starting comment a little to make it hopefully a little better to understand what I mean.
- Added os specific files to avoid thumbnails and similar things to be committed.
- Added the line you proposed.

What do you think?

Pfeil commented:
If this is fine, I think you'll have to merge, I think I can not ... probably that is why I forked the template repo back then :)

Pfeil commented:
Merging the master has deleted the travis.yaml, still it seems to be triggered? The rest of the CI is probably fixed by #6 by then.

VolkerHartmann commented on .gitignore:
```diff
@@ -1,9 +1,131 @@
+# Ignore Gradle GUI config
+gradle-app.setting
+# Avoid ignoring Gradle wrapper jar file (.jar files might be ignored)
+!gradle-wrapper.jar
```
Works not for me!?
!gradle/wrapper/gradle-wrapper.jar may be better.

VolkerHartmann replied:
Alternatively to make it more flexible:
!**/gradle-wrapper.jar

Pfeil replied:
Whoops, thanks. Should be fixed now.

VolkerHartmann commented on .gitignore:
```diff
@@ -1,9 +1,150 @@
+!src/**/build/
+# Avoid ignoring Gradle wrapper jar file (.jar files might be ignored)
+!**/gradle-wrapper.jar
+##############
```
New line missing at the end!

Pfeil replied:
Done :)
